server/db.go

```diff
@@ -448,16 +442,16 @@ func (c *Server) fetchIssuerByType(ctx context.Context, issuerType string) (*Iss
 	}
 
 	if c.caches != nil {
-		c.caches["issuer"].SetDefault(issuerType, issuerV3)
+		c.caches["issuer"].SetDefault(issuerType, &issuerV3)
 	}
 
 	return convertedIssuer, nil
 }
 
-func (c *Server) fetchIssuers(issuerType string) (*[]Issuer, error) {
-	if c.caches != nil {
-		if cached, found := c.caches["issuers"].Get(issuerType); found {
-			return cached.(*[]Issuer), nil
+func (c *Server) fetchIssuers(issuerType string) ([]Issuer, error) {
+	if cached := retrieveFromCache(c.caches, "issuers", issuerType); cached != nil {
+		if issuers, ok := cached.([]Issuer); ok {
+			return issuers, nil
```

Comment:
One other thing here:

```
var nilslc []string
fmt.Println([]string{} == nil) // false
fmt.Println(nilslc == nil) //true
```

I have a feeling we're not going to cache an empty slice, but it's not the same as a nil slice, so the check `cached != nil` could allow the return of an empty slice.

Reply:
This function really will only ever return a slice of 1 issuer. I think we can allow this through and come back to refactor the slice issue generally.